tests/unit/test_credentials.py

```diff
@@ -972,5 +973,137 @@ def test_source_profile_does_not_exist(self):
             provider.load()
 
 
+class TestSamlGenericFormsBasedAuthenticator(unittest.TestCase):
+    authenticator = credentials.SamlGenericFormsBasedAuthenticator()
+    login_form = '''<html><form action="login">
+        <div>The &nbsp; in the form will not confuse our parser.</div>
+        <input name="foo" value="bar"/>
+        <input name="username"/><input name="password"/>
+        </form></html>'''
+    GET = 'botocore.credentials.requests.get'
+    POST = 'botocore.credentials.requests.post'
+    profile = {
+        'saml_endpoint': 'http://notexist.com',
+        'saml_authentication_type': 'form',}
+
+    @mock.patch(GET, return_value=mock.Mock(text='<html>wrong way</html>'))
+    def test_login_form_not_exist(self, _get):
+        with self.assertRaises(botocore.exceptions.SamlError):
+            if self.authenticator.is_suitable(self.profile):
+                self.authenticator.authenticate(
+                    self.profile, lambda prompt: 'foo', lambda prompt: 'bar')
+
+    @mock.patch(POST, return_value=mock.Mock(text='<html>login failed</html>'))
+    @mock.patch(GET, return_value=mock.Mock(text=login_form))
+    def test_no_saml_assertion(self, _get, _post):
```

Comment:
If you don't need the value provided in a mock.patch you can add the mock value as an argument to patch().  Then the decorated function won't be passed in the arguments.